elements/bulbs-carousel/bulbs-carousel.js

```diff
@@ -0,0 +1,174 @@
+import invariant from 'invariant';
+import { BulbsHTMLElement, registerElement } from 'bulbs-elements/register';
+import { moveChildren } from 'bulbs-elements/util';
+
+import './bulbs-carousel.scss';
+
+export const supportsCalcInTransform = (() => {
+  let div = document.createElement('div');
+  div.style.transform = 'translateX(calc(100% - 10px))';
+  return div.style.transform !== '';
+})();
+
+export default class BulbsCarousel extends BulbsHTMLElement {
```

Comment:
I walked down the road of something like:

```js
class BulbsCarousel extends BulbsHTMLElement {
  createdCallback () {
    /* ... */
    this.state = new BulbsCarouselState({
      carouselItems: this.track.children,
      currentIndex: 0,
    });
  }
}
```

`BulbsCarouselState` carried all the `get*` methods, `updateCurrentIndex`, `updateCurrentIndex`, `pageToCarouselItem`, `slideToPrevious`, `slideToNext`, `isOnFirstPage`, and `isOnLastPage`.

`BulbsCarousel` keeps: `createdCallback`, `attachedCallback`, `handleClick`, and `applyState`.

External access to `state` would be discouraged.